apps/site/pages/en/learn/test-runner/mocking.md

```diff
@@ -0,0 +1,270 @@
+---
+title: Mocking in tests
+layout: learn
+authors: JakobJingleheimer
+---
+
+# Mocking in tests
+
+Tests should be deterministic: runnable in any order, any number of times, and always produce the same result. Proper setup and mocking make this possible.
+
+Mocking is a means of creating a facsimile, a puppet. This is generally done in a `when 'a', do 'b'` manner of puppeteering. The idea is to limit the number of moving pieces and control things that "don't matter". "mocks" and "stubs" are technically different kinds of "test doubles". For the curious mind, a stub is a replacement that does nothing (a no-op) but track its invocation. A mock is a stub that also has a fake implementation (the `when 'a', do 'b'`). Within this doc, the difference is unimportant, and stubs are referred to as mocks.
```

Comment:
```suggestion
Mocking is a mean of creating a facsimile, a puppet. This is generally done in a `when 'a', do 'b'` manner of puppeteering. The idea is to limit the number of moving pieces and control things that "don't matter". "mocks" and "stubs" are technically different kinds of "test doubles". For the curious mind, a stub is a replacement that does nothing (a no-op) but track its invocation. A mock is a stub that also has a fake implementation (the `when 'a', do 'b'`). Within this doc, the difference is unimportant, and stubs are referred to as mocks.
```

Reply:
I think it is correct as written: https://dictionary.cambridge.org/dictionary/english/means

Reply:
It seems the UK version takes an `s` as you linked, but IIUC the US version doesn't (e.g. https://en.wiktionary.org/wiki/mean#Noun), and I think we tend to follow US conventions.

Reply:
its correct as `means`